Firestore/core/src/firebase/firestore/remote/serializer.h

```diff
@@ -360,7 +366,6 @@ class Serializer {
       const google_firestore_v1_ExistenceFilter& filter) const;
 
   model::DatabaseId database_id_;
-  const std::string database_name_;
```

Comment:
Objective-C code [always reencodes](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/Source/Remote/FSTSerializerBeta.mm#L242) database ID, while Android [does it once](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteSerializer.java#L114) and [caches the value](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteSerializer.java#L241).

I removed this member variable because it was unused. Alternatively, I can implement caching similar to Android.

Reply:
It's hard to cache in a useful way on iOS because of the raw pointers in use in nanopb. Not caching is probably preferable until we can assign values to nanopb messages without giving ownership.